src/mantine-hooks/src/use-timeout/use-timeout.ts

```diff
@@ -7,29 +7,32 @@ export function useTimeout(
 ) {
   const timeoutRef = useRef<number | null>(null);
 
-  const start = (...callbackParams: any[]) => {
-    if (!timeoutRef.current) {
-      timeoutRef.current = window.setTimeout(() => {
-        callback(callbackParams);
-        timeoutRef.current = null;
-      }, delay);
-    }
-  };
+  const start = useCallback(
+    (...callbackParams: any[]) => {
+      if (!timeoutRef.current) {
+        timeoutRef.current = window.setTimeout(() => {
+          callback(callbackParams);
+          timeoutRef.current = null;
+        }, delay);
+      }
+    },
+    [callback, delay],
+  );
 
-  const clear = () => {
+  const clear = useCallback(() => {
     if (timeoutRef.current) {
       window.clearTimeout(timeoutRef.current);
       timeoutRef.current = null;
     }
-  };
+  }, []);
 
   useEffect(() => {
     if (options.autoInvoke) {
       start();
     }
 
     return clear;
-  }, [delay]);
+  }, [clear, delay, options.autoInvoke, start]);
```

Comment:
There is no reason to add options.autoInvoke to the dependencies array. `start` should be called exactly once when hook is initialized.

Reply:
@Raicuparta you did not resolve my comment.

Reply:
Do we really want an effect here then?

The way it was before, changing the delay would also autoInvoke again, but changing the callback wouldn't. Was that intended? Because it seems very counterintuitive to me.

In this PR, changing the callback, the delay, or the autoInvoke option would trigger the autoInvoke again.

If the intention is to really only make it run exactly once, we should just have no dependencies at all, but then maybe it shouldn't be in an effect at all.

I have removed autoInvoke from the dependencies anyway to avoid making changes outside the scope of the PR, but this is pretty clearly against the intended useEffect usage as per the docs: https://react.dev/learn/removing-effect-dependencies.

(I also removed delay from the dependencies since it is now redundant)

Reply:
Oh and yeah I didn't mean to make that first push that only removed the delay, sorry about the notification spam